Add `lax`/`strict` values to `fix` option and CLI flag (#8106)
stylelint/stylelint: docs/user-guide/options.md

Automatically fix, where possible, problems reported by rules.

Options are:

> - `true` or `lax` - uses [postcss-safe-parser](https://www.npmjs.com/package/postcss-safe-parser) to fix as much as possible, even when there are syntax errors
> - `strict` - uses [PostCSS Parser](https://postcss.org/api/#postcss-parser) and only fixes problems when there are no syntax errors

Reviewer: [suggestion] To distinguish from boolean values, how about adding quotes like this?

```suggestion
- `true` or `"lax"` - uses [postcss-safe-parser](https://www.npmjs.com/package/postcss-safe-parser) to fix as much as possible, even when there are syntax errors
- `"strict"` - uses [PostCSS Parser](https://postcss.org/api/#postcss-parser) and only fixes problems when there are no syntax errors
```

- `false` (default) - does not fix any problems

> 

Reviewer: [suggestion] This is a just idea and not required, but what about adding the following tip?

> [!TIP]
> If you don't care about strictness now, we recommend using `true` because `"strict"` may become the default in the future.

```suggestion

> [!TIP]
> If you don't care about strictness now, we recommend using `true` because `"strict"` may become the default in the future.

```

Reviewer 2: refs
stylelint/stylelint.io#409
https://github.com/incentro-ecx/remark-github-admonitions-to-directives#%EF%B8%8F-customizing-the-mapping

Automatically fix, where possible, problems reported by rules.

Options are:

- `"lax"` (default) - uses [postcss-safe-parser](https://www.npmjs.com/package/postcss-safe-parser) to fix as much as possible, even when there are syntax errors
- `"strict"` - uses [PostCSS Parser](https://postcss.org/api/#postcss-parser) and only fixes problems when there are no syntax errors

> [!TIP]

> > If you don't care about strictness now, we recommend using `--fix` because `"strict"` may become the default in the future.

Reviewer 2: ```suggestion
> If you don't care about strictness now, we recommend using the boolean flag `--fix`—which _currently_ maps to `--fix=lax`—without any value because `"strict"` may become the new default in the future.
```

> > If you don't care about strictness now, we recommend using `--fix` because `"strict"` may become the default in the future.

Author: Thank you both for your suggestions. 
Since it says `"lax" (default)`, I thought it’s unnecessary to mention that it’s mapped!

> > If you don't care about strictness now, we recommend using `--fix` without any option because `"strict"` may become the default in the future.

Reviewer 3: ```suggestion
> We recommend using `--fix=strict` if you are linting a language extension like SCSS
```
